### ISSUE TITLE
V3.1.3: Panics when launching

### ISSUE THREAD

**Issue by Schievel1** [an-anime-team/an-anime-game-launcher#94]
Got this when I try to start version 3.1.3.

```
~ ✗ RUST_BACKTRACE=1 /usr/bin/anime-game-launcher                                                      took 253ms

(process:25779): Gtk-WARNING **: 20:02:39.396: Theme parser error: gtk.css:251:11-12: Expected a filter

(process:25779): Gtk-WARNING **: 20:02:39.410: Theme parser error: gtk-dark.css:251:11-12: Expected a filter

(process:25779): Gtk-WARNING **: 20:02:39.427: Theme parser error: gtk-dark.css:251:11-12: Expected a filter

(process:25779): Gtk-WARNING **: 20:02:39.441: Theme parser error: gtk-dark.css:251:11-12: Expected a filter
thread 'main' panicked at 'byte index 6 is out of bounds of ``', src/ui/about.rs:83:66
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

3.0.1 was working fine.

**Comment by Mar0xy** [comment]
Is this on a fresh install of the launcher?
Also what's the GTK Version installed as on my system I don't seem to have this issue same for the flatpak it works fine

**Comment by Schievel1** [comment]
Yup, purged any previous install and tried again, same result. 
Gtk 3/4 versions below. 

I think this has something to do with how I build this. It panics on the line
https://github.com/an-anime-team/an-anime-game-launcher/blob/095966d45133352c66b8f61ab631847412fc6882/src/ui/about.rs#L83

This commit hash str comes from the sdk, here: https://github.com/an-anime-team/anime-launcher-sdk/blob/6bf720a1571ff7462e09e7ca2c0145d07933e3e4/src/lib.rs#L27

Which is getting it from the environment variable. (I guess cargo is setting that variable during build, because I can not find it when I am in a git repo.)

The ebuild does not check out the git repos recursively like you maybe do in the flatpak because versioned ebuilds do not support this. Instead I download a tarball via the commit hash and link it into the right place in the repo.

My only hope is making a live ebuild now. I don't feel like patching this out because I can already barely keep up with you guys pace. 

```
*  gui-libs/gtk
      Latest version available: 4.10.0
      Latest version installed: 4.10.0
      Size of files: 19.992 KiB
      Homepage:      https://www.gtk.org/ https://gitlab.gnome.org/GNOME/gtk/
      Description:   GTK is a multi-platform toolkit for creating graphical user interfaces
      License:       LGPL-2+

*  x11-libs/gtk+
      Latest version available: 3.24.37
      Latest version installed: 3.24.37
      Size of files: 12.111 KiB
      Homepage:      https://www.gtk.org/
      Description:   Gimp ToolKit +
      License:       LGPL-2+
```

**Comment by Mar0xy** [comment]
I am actually gonna reopen this as this bug also affects the RPM builds which is bizzare as the binary itself works when downloaded from github so it must be GTK version related

**Comment by Mar0xy** [comment]
So this about git hash stuff has to generally be removed as it just won't play nice with packaging on a lot of services and in my honest opinion not needed anyways as the version + tag is enough to lead back to commit history which was Krypton's point.

In fact he also thought about replacing version with hashes fully which I am fully against.

**Comment by krypt0nn** [comment]
Fixed in 3.1.4: https://github.com/an-anime-team/an-anime-game-launcher/commit/58743eca04539e5d3cfd24baf7c78e4ec3686799
